File: Movie/Models/MovieDetail.swift
//
//  MovieDetail.swift
//  Movie
//
//

import Foundation

struct MovieDetail: Codable, Identifiable  {
    let id: Int
    let title: String
    let genres: [Genre]
    let backdropPath: String?
    let overview: String
    let releaseDate: Date
    let voteAverage: Double
    
    var backdropURL: URL {
        return URL(string: "https://image.tmdb.org/t/p/original\(backdropPath ?? "")")!
    }
    
    var voteAverageText: String {
        return String(format: "%.1f", voteAverage)
    }
    
    enum CodingKeys: String, CodingKey {
        case id
        case title
        case genres
        case backdropPath = "backdrop_path"
        case overview
        case releaseDate = "release_date"
        case voteAverage = "vote_average"
    }
}

struct Genre: Codable, Identifiable {
    let id: Int
    let name: String
}

File: shared/src/commonMain/kotlin/com/mehmetaliocak/movie_app/models/MovieDetail.kt
package com.mehmetaliocak.movie_app.models

import com.mehmetaliocak.movie_app.util.asPhotoUrl
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MovieDetail(
    val id: Long,
    val title: String,
    val genres: List<Genre>,
    @SerialName("backdrop_path")
    private val backdropPath: String?,
    val overview: String,
    @SerialName("release_date")
    val releaseDate: String?,
    @SerialName("vote_average")
    val voteAverage: Double,
) {
    fun backdropURL(): String {
        return backdropPath.asPhotoUrl()
    }
}